riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/DefaultRiptideConfigurer.java

```diff
@@ -0,0 +1,99 @@
+
+import static org.zalando.riptide.autoconfigure.ValueConstants.LOGBOOK_REF;
+import static org.zalando.riptide.autoconfigure.ValueConstants.METER_REGISTRY_REF;
+import static org.zalando.riptide.autoconfigure.ValueConstants.TRACER_REF;
+
+@AllArgsConstructor
+class DefaultRiptideConfigurer {
+    private final ConfigurableListableBeanFactory beanFactory;
+    private final RiptideProperties properties;
+
+    void register() {
+        properties.getClients().forEach(this::configure);
+    }
+
+    private void configure(final String id, final RiptideProperties.Client client) {
+        if (client.getTracing().getEnabled()) {
+            final BeanDefinition tracerRef = getBeanRef(Tracer.class);
+
+            findBeanDefinition(id, TracedExecutorService.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, TRACER_REF, tracerRef));
+
+            findBeanDefinition(id, OpenTracingPlugin.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, TRACER_REF, tracerRef));
+
+            findBeanDefinition(id, TracedScheduledExecutorService.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, TRACER_REF, tracerRef));
+        }
+
+        if (client.getLogging().getEnabled()) {
+            final BeanDefinition logbookRef = getBeanRef(Logbook.class);
+
+            findBeanDefinition(id, LogbookPlugin.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, LOGBOOK_REF, logbookRef));
+        }
+
+        if (client.getMetrics().getEnabled()) {
+            final BeanDefinition meterRegistryRef = getBeanRef(MeterRegistry.class);
+
+            findBeanDefinition(id, MetricsPlugin.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, METER_REGISTRY_REF, meterRegistryRef));
+
+            findBeanDefinition(id, RetryListener.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, METER_REGISTRY_REF, meterRegistryRef));
+
+            findBeanDefinition(id, CircuitBreakerListener.class)
+                    .ifPresent(bd -> replaceConstructorArgumentWithBean(bd, METER_REGISTRY_REF, meterRegistryRef));
+        }
+    }
+
+    private BeanDefinition getBeanRef(Class type) {
+        final String[] names = beanFactory.getBeanNamesForType(type);
+        BeanDefinition definition = null;
+        for (final String name : names) {
+            definition = beanFactory.getBeanDefinition(name);
+            if (definition.isPrimary()) {
+                return definition;
+            }
+        }
+        return definition;
+    }
+
+    private Optional<BeanDefinition> findBeanDefinition(final String id, final Class<?> type) {
+        return findBeanDefinition(Name.name(id, type));
+    }
+
+    private Optional<BeanDefinition> findBeanDefinition(final Name name) {
+        try {
+            return Optional.of(beanFactory.getBeanDefinition(name.toNormalizedString()));
```

Comment:
Shouldn't this search for all possible spellings (`getAlternatives`)?

Reply:
Don't think so. 
Alternative names can appear from plugins that we're overwritten by user, in such case it would be a bit unfair to mess with user configuration. And if bean was registered by the registar - the normalisedname is used.

Reply:
Right